wizard.py

```diff
@@ -0,0 +1,1037 @@
+
+
+def _persist_session_subdocument_ts(subdocument):
+  """Updates/adds last_modified to the subdocument before persisting it. """
+
+  for key in subdocument.keys():
+    subdocument[key]["last_modified"] = time.time()
+  _persist_session_subdocument(subdocument)
+
+
+def _get_session_subdocument(key):
+  """Returns a subdocument (e.g. vcs, ssc, functionaries...) identified by
+  passed key from session document identified by current session id.
+  Returns an empty dict if document or subdocument are not found.  """
+  if not session.get("id"):
+    abort(404)
+
+    # Get session document (use short circuit for default empty dict)
+  session_doc = mongo.db.session_collection.find_one(
+      {"_id": session["id"]})
+
+  if not session_doc:
+    return {}
+
+  # Get vcs data from session document or empty dict
+  return session_doc.get(key, {})
+
+
+def _get_session_document():
+  """Returns the entire session document.
+  Returns an empty dict if document or subdocument are not found.  """
+
+  if not session.get("id"):
+    abort(404)
+
+  session_doc = mongo.db.session_collection.find_one(
+    {"_id": session["id"]})
+
+  if not session_doc:
+    return {}
+
+  return session_doc
+
+
+# -----------------------------------------------------------------------------
+# View Decorator & Hooks
+# -----------------------------------------------------------------------------
+def with_session_id(wrapped_func):
+  """
+  Generate new session id if it does not exist.
+
+  For now, a user could start a new session on any page
+  TODO: Should we redirect to the start page if the session is new?
+  """
+  @wraps(wrapped_func)
+  def decorated_function(*args, **kwargs):
+    if not session.get("id"):
+      # Security is not paramount, we don't store sensitive data, right?
+      session["id"] = uuid.uuid4()
+      app.logger.info("New session ID '{}'".format(session["id"]))
+
+    return wrapped_func(*args, **kwargs)
+  return decorated_function
+
+
+@app.after_request
+def ajax_flash_messages(response):
+  """ This function intercepts JSON responses to ajax requests and injects
+  a "messages" field containing flashed messages.
+  To display them the JS callback that receives the response can call
+  show_messages(repsonse.messages).
+  """
+
+  if (request.is_xhr and
+      response.headers.get("Content-Type") == "application/json"):
+    response_data = json.loads(response.get_data())
+    response_data["messages"] = get_flashed_messages(with_categories=True)
+    response.set_data(json.dumps(response_data))
+
+  return response
+
+
+# -----------------------------------------------------------------------------
+# Views
+# -----------------------------------------------------------------------------
+@app.route("/")
+@with_session_id
+def start():
+  """Step 0.
+  Wizard entry point, static landing page. """
+  return render_template("start.html")
+
+
+@app.route("/versioning", methods=["GET", "POST"])
+@with_session_id
+def versioning():
```

Comment:
Not a big deal, but at first I thought this was a page about [versioning software](https://en.wikipedia.org/wiki/Software_versioning).  What about just `vcs`, it's concise and more obvious.